### PR TITLE
update Vue.js to 1.0.0

### PR DESCRIPTION
**Pull request by yyx990803**
Ran the tests and all passed.

### CODE REVIEW

**Comment by samccone** [comment]
Hi @yyx990803 thanks for opening the PR so quickly after the release (congrats on the PS)

screen shot 2015-10-27 at 9 33 30 am

Looks like you have a few style issues to fixup before we can merge, thanks!

**Comment by yyx990803** [comment]
Thanks! Should be fixed now.

File: examples/vue/js/app.js [hunk 1]
```diff
@@ -117,6 +105,21 @@
...
+		// a custom directive to wait for the DOM to be updated
+		// before focusing on the input field.
+		// http://vuejs.org/guide/custom-directive.html
+		directives: {
+			'todo-focus': function (value) {
+				if (!value) {
+					return;
+				}
+				var el = this.el;
+				Vue.nextTick(function () {
```
Comment (samccone) [inline examples/vue/js/app.js]: could this be?
`Vue.nextTick(this.el.focus)`

File: examples/vue/js/app.js [hunk 2]
```diff
@@ -33,25 +33,13 @@
...
-		// http://vuejs.org/guide/directives.html#Writing_a_Custom_Directive
-		directives: {
-			'todo-focus': function (value) {
-				if (!value) {
-					return;
+		// watch todos change for localStorage persistence
+		watch: {
+			todos: {
+				deep: true,
+				handler: function (todos) {
```
Comment (samccone) [inline examples/vue/js/app.js]: could this be 

```js
handler: todoStorage.save
```

?

Comment (yyx990803) [inline examples/vue/js/app.js]: yeah good idea

File: examples/vue/js/app.js (hunk 1 above)
Comment (yyx990803) [inline examples/vue/js/app.js]: this would result in Illegal invocation. `focus` must be invoked as a method on the element.

Comment (samccone) [inline examples/vue/js/app.js]: word, was not sure what the context was here, does `nextTick` take an optional this context argument?

Comment (yyx990803) [inline examples/vue/js/app.js]: yes, although imo it doesn't look much better:

``` js
Vue.nextTick(this.el.focus, this.el)
```

Comment (samccone) [inline examples/vue/js/app.js]: ok no worries just asking:)

**Comment by samccone** [comment]
:+1: LGTM 

R: @arthurvr @addyosmani

**Pull request merged**